pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -277,6 +286,10 @@ type Controller struct {
 
 	podsClient kclient.Client[*v1.Pod]
 
+	workloadentries kclient.Client[*v1alpha3.WorkloadEntry]
```

Comment:
I think this belongs in autoregistration, not here for few reasons:

1. Auto registration is flag controlled; this is not, so we get unexpected enablement
2. This usage doesn't properly respect the semantics of Istio CRD reading. There is some work (https://github.com/istio/istio/pull/44145 is a start ,long way to go) to unify this with other k8s types, but still some major differences in handling the possible absence of the CRD, reading from other sources (files/xds), and respecting istio.io/rev labelling.
3. A bit confusing to have this logic separate from the rest of auto-registration.

In autoregistration we already have a `store` passed in which we can use to add an event handler to.

Reply:
sg, fixed